### PR TITLE
Some simple doc fixes

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -660,37 +660,39 @@ The `meson` object allows you to introspect various properties of the system. Th
...
+- `has_exe_wrapper()` returns true when doing a cross build if there is a wrapper command that can be used to execute cross built binaries (for example when cross compiling from Linux to Windows, one can use `wine` as the wrapper).
 
 - `add_install_script(script_name, arg1, arg2, ...)` causes the script given as an argument to be run during the install step, this script will have the environment variables `MESON_SOURCE_ROOT`, `MESON_BUILD_ROOT`, `MESON_INSTALL_PREFIX`, and `MESON_INSTALL_DESTDIR_PREFIX` set. All additional arguments are passed as parameters.
 
-  `MESON_INSTALL_PREFIX` has the value of the `prefix` option passed to Meson, and `MESON_INSTALL_DESTDIR_PREFIX` has both the value of `DESTDIR` and `prefix` joined together. This is useful because both are absolute paths, and many path-joining functions such as [`os.path.join` in Python](https://docs.python.org/3/library/os.path.html#os.path.join) special-case absolute paths.
+  `DESTDIR` will be passed through to the script if specified.
```
Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: This is a bit misleading because it implies that we filter some environment variables. We don't.

File: docs/markdown/Reference-manual.md [hunk 2]
```diff
@@ -660,37 +660,39 @@ The `meson` object allows you to introspect various properties of the system. Th
...
+
+  `MESON_INSTALL_PREFIX` has the value of the `prefix` option passed to Meson, and `MESON_INSTALL_DESTDIR_PREFIX` contains `DESTDIR` and `prefix` joined together. This is useful because both are absolute paths, and many path-joining functions such as [`os.path.join` in Python](https://docs.python.org/3/library/os.path.html#os.path.join) special-case absolute paths.
 
 - `add_postconf_script(script_name, arg1, arg2, ...)` will run the executable given as an argument after all project files have been generated. This script will have the environment variables `MESON_SOURCE_ROOT` and `MESON_BUILD_ROOT` set.
 
 - `current_source_dir()` returns a string to the current source directory. Note: **you do not need to use this function** when passing files from the current source directory to a function since that is the default. Also, you can use the `files()` function to refer to files in the current or any other source directory instead of constructing paths manually with `meson.current_source_dir()`.
 
-- `current_build_dir()` returns a string to the current build directory
+- `current_build_dir()` returns a string witht the absolute path to the current build directory.
```
Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: Typo

File: docs/markdown/Reference-manual.md [hunk 3]
```diff
@@ -741,6 +743,7 @@ This object is returned by [`meson.get_compiler(lang)`](#meson-object). It repre
 
 - `get_id()` returns a string identifying the compiler (e.g. `'gcc'`)
 - `version()` returns the compiler's version number as a string
+- `cmd_array()` returns the command to invoke the compiler as a list (e.g. `['ccache', 'cc']`), available since 0.24.0
```
Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: This function was added for use in tests, and is not documented on purpose. It is our concern that people will look at the compiler object, find this, and start using it to compile everything instead of using `executable()` or `library()`, etc. That would be a terrible situation because it means that your compiled objects won't integrate with the project options, default compiler args, and so on.

If we do document this, I think we should add some text that makes it clear that this is supposed to be used for edge-cases, and that people should report their edge-cases to us so we can figure out a way to support it without them having to resort to using this.

Comment (QuLogic) [inline docs/markdown/Reference-manual.md]: Or if you don't document, their edge cases can't be hacked around and they'll actually report it instead of hacking around it and forgetting to report anything.

Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: Exactly! Strongly encouraging users to actually report deficiencies is really important.

Comment (TingPing) [inline docs/markdown/Reference-manual.md]: Same comment as elsewhere, prefix these with `_` and enforce they are only called from within meson tests?

Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: In this case, this feature is used by systemd for an edge-case which can't be done in any other way, and such edge-cases will always exist, IMO.

**Comment by keszybz** [comment]
Updated to address the comments. I don't believe in "hiding" stuff, so I think it's better to describe the functionality, and instead make a note why it should not be used, unless there's a good reason.

**Comment by jpakkane** [comment]
> I don't believe in "hiding" stuff, so I think it's better to describe the functionality

The correct solution is to add whatever functionality is missing so systemd can do the test properly and then remove this. The fact that it is even possible to invoke is a severe bug.

**Comment by keszybz** [comment]
Making `cc` a normal command that can be embedded anywhere sounds like a good plan. I'll repush the commits without any mention of `cmd_array`.
